### DISCUSSION TITLE
QST: Should geopandas use shapely to identify GEOS version?

### DISCUSSION THREAD

**Discussion by dazza-codes** [geopandas/geopandas#2707] [Q&A]
This is not a usage related question, it's an implementation question for geopandas

#### Question about geopandas

Q.  On the same venv, why does `geopandas` and `shapely` not share the same `libgeos` version? e.g.

```python
>>> import geopandas as gpd
>>> gpd.show_versions()

SYSTEM INFO
-----------
python     : 3.7.10 | packaged by conda-forge | (default, Feb 19 2021, 16:07:37)  [GCC 9.3.0]
executable : /opt/conda/envs/gis/bin/python
machine    : Linux-4.15.0-142-generic-x86_64-with-debian-buster-sid

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.7.1
GEOS lib   : /usr/lib/x86_64-linux-gnu/libgeos_c.so
GDAL       : 3.2.2
GDAL data dir: /opt/conda/envs/gis/lib/python3.7/site-packages/fiona/gdal_data
PROJ       : 7.0.1
PROJ data dir: /opt/conda/envs/gis/lib/python3.7/site-packages/pyproj/proj_dir/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.9.0
pandas     : 1.2.4
fiona      : 1.8.19
numpy      : 1.20.2
shapely    : 1.7.1
rtree      : 0.9.7
pyproj     : 2.6.1.post1
matplotlib : 3.4.1
mapclassify: None
geopy      : 2.1.0
psycopg2   : 2.8.6 (dt dec pq3 ext lo64)
geoalchemy2: 0.8.5
pyarrow    : 3.0.0
pygeos     : None

>>> import shapely.geos
>>> shapely.geos._lgeos
<CDLL '/opt/conda/envs/gis/lib/python3.7/site-packages/shapely/.libs/libgeos_c-a68605fd.so.1.13.1', handle 55a470a1fc30 at 0x7f30bbdf6fd0>
>>> 
```

Is `shapely` exposing some other `libgeos` version to `geopandas` or does the latter side-step `shapely` entirely to go find a different `libgeos` installation?

On another system (venv) where there is no system-wide installation for `libgeos`, `geopandas` reports there is no `GEOS`, but `shapely` is working fine with a bundled wheel for `libgeos`.

Should `geopandas` use `shapely.geos` to identify `libgeos` version details?, e.g.

```python
>>> shapely.geos.geos_version
(3, 8, 0)
>>> shapely.geos.geos_capi_version
(1, 13, 1)
>>> shapely.geos.geos_version_string
'3.8.0-CAPI-1.13.1 '
```

**Comment by jorisvandenbossche** [comment]
GeoPandas does not actually use GEOS directly, but only through Shapely (or pygeos), so by definition it's always using the GEOS that is used by Shapely.

Now, why we report something differently in `geopandas.show_versions()` is something else. We are actually using Shapely to get this information:

https://github.com/geopandas/geopandas/blob/f4a9763f90d65dea21e2e628e60326e27ca13576/geopandas/tools/_show_versions.py#L46-L49

But so we might be using the "wrong" way to get this information, as it seems different from `shapely.geos` in your case.

**Comment by jorisvandenbossche** [comment]
So we should probably use `shapely.geos` instead of `shapely._buildcfg` to determine the GEOS version Shapely is using.
